libs/langchain/langchain/vectorstores/redis/schema.py

```diff
@@ -0,0 +1,276 @@
+    sortable: Optional[bool] = False
+
+    def as_field(self) -> TagField:
+        return TagField(
+            self.name,
+            separator=self.separator,
+            case_sensitive=self.case_sensitive,
+            sortable=self.sortable,
+            no_index=self.no_index,
+        )
+
+
+class NumericFieldSchema(RedisField):
+    no_index: bool = False
+    sortable: Optional[bool] = False
+
+    def as_field(self) -> NumericField:
+        return NumericField(self.name, sortable=self.sortable, no_index=self.no_index)
+
+
+class RedisVectorField(RedisField):
+    dims: int = Field(...)
+    algorithm: object = Field(...)
+    datatype: str = Field(default="FLOAT32")
+    distance_metric: RedisDistanceMetric = Field(default="COSINE")
+    initial_cap: int = Field(default=20000)
+
+    @validator("distance_metric", pre=True)
+    def uppercase_strings(cls, v: str) -> str:
+        return v.upper()
+
+    @validator("datatype", pre=True)
+    def uppercase_and_check_dtype(cls, v: str) -> str:
+        if v.upper() not in REDIS_VECTOR_DTYPE_MAP:
+            raise ValueError(
+                f"datatype must be one of {REDIS_VECTOR_DTYPE_MAP.keys()}. Got {v}"
+            )
+        return v.upper()
+
+
+class FlatVectorField(RedisVectorField):
+    algorithm: Literal["FLAT"] = "FLAT"
+    block_size: int = Field(default=1000)
+
+    def as_field(self) -> VectorField:
+        return VectorField(
+            self.name,
+            self.algorithm,
+            {
+                "TYPE": self.datatype,
+                "DIM": self.dims,
+                "DISTANCE_METRIC": self.distance_metric,
+                "INITIAL_CAP": self.initial_cap,
+                "BLOCK_SIZE": self.block_size,
+            },
+        )
+
+
+class HNSWVectorField(RedisVectorField):
+    algorithm: Literal["HNSW"] = "HNSW"
+    m: int = Field(default=16)
+    ef_construction: int = Field(default=200)
+    ef_runtime: int = Field(default=10)
+    epsilon: float = Field(default=0.8)
+
+    def as_field(self) -> VectorField:
+        return VectorField(
+            self.name,
+            self.algorithm,
+            {
+                "TYPE": self.datatype,
+                "DIM": self.dims,
+                "DISTANCE_METRIC": self.distance_metric,
+                "INITIAL_CAP": self.initial_cap,
+                "M": self.m,
+                "EF_CONSTRUCTION": self.ef_construction,
+                "EF_RUNTIME": self.ef_runtime,
+                "EPSILON": self.epsilon,
+            },
+        )
+
+
+class RedisModel(BaseModel):
+    # always have a content field for text
+    text: List[TextFieldSchema] = [TextFieldSchema(name="content")]
+    tag: Optional[List[TagFieldSchema]] = None
+    numeric: Optional[List[NumericFieldSchema]] = None
+    extra: Optional[List[RedisField]] = None
+
+    # filled by default_vector_schema
+    vector: Optional[List[Union[FlatVectorField, HNSWVectorField]]] = None
+    content_key: str = "content"
```

Comment:
If I understand correctly, having a field called "content" is mandatory.
Fix me if I'm wrong but the only location I found it might be a problem to give this field a user-defined name is in `similarity_search_with_score` where we return `result.content` 
(in `similarity_search()` it is handled better IMO by using `getattr(result, content_key)` instead)
Another solution is to define the return fields as
`Query().return_field(self.content_field, as_field="content")`

Also, where is the field name enforced in `from_existing_index()`?